numba/targets/arraymath.py

```diff
@@ -1414,6 +1418,34 @@ def np_tril_impl_multi(m, k=0):
         return np_tril_impl_multi
 
 
+@overload(np.tril_indices)
+def np_tril_indices(n, k=0, m=None):
+
+    # we require integer arguments, unlike numpy
+    _check_is_integer(n, 'n')
+    _check_is_integer(k, 'k')
+    if m is not None:
```

Comment:
This is a `numba.numpy_support.is_nonelike` case, None literal vs NoneType. Merge in master to get that function from it's new API home.

This ought to work:
```python
from numba import njit
import numpy as np


@njit
def triu_indices_n(n, k=0, m=None):
    return np.triu_indices(n, k, m)

print(triu_indices_n(2))
print(triu_indices_n.py_func(2))
```
something like (but with the import at the top of file) will likely sort it:
```python
    from numba.numpy_support import is_nonelike
    if not is_nonelike(m):
        _check_is_integer(m, 'm')
```
same argument applies to the `tril` case.

Reply:
yes, the tests don't actually capture this case..